### PR TITLE
Fix more compiler warnings

### PR DESCRIPTION
**Pull request by sicherha**
In the long run, I would like to enable `-Wall -Wextra` by default to catch such issues in advance. Unfortunately, we're not there yet - among other things, GCC gives a bunch of warnings about calling `memset()` to initialise a non-trivial object type.

### CODE REVIEW

File: mold.h [hunk 1]
```diff
@@ -311,7 +311,7 @@ inline u64 read_uleb(u8 const*&buf) {
 }
 
 inline i64 uleb_size(u64 val) {
-#pragma unroll
+#pragma GCC unroll 8
```
Comment (rui314) [inline mold.h:314]: This is a valid pragma for Clang.

https://clang.llvm.org/docs/AttributeReference.html#pragma-unroll-pragma-nounroll

File: cmdline.h [hunk 2]
```diff
@@ -61,7 +61,7 @@ std::vector<std::string_view> expand_response_files(C &ctx, char **argv) {
   return vec;
 }
 
-static std::string_view string_trim(std::string_view str) {
+[[maybe_unused]] static std::string_view string_trim(std::string_view str) {
```
Comment (rui314) [inline cmdline.h:64]: Instead of adding an attribute, can you change this to `static inline`?

Comment (sicherha) [reply cmdline.h]: Done.

File: mold.h (hunk 1 above)
Comment (sicherha) [reply mold.h:314]: `#pragma GCC unroll <n>` is supported by both GCC and Clang, whereas `#pragma unroll` is only supported by Clang. Using the GCC-compatible notation fixes an *unknown pragma* warning when compiling with GCC.

**Comment by rui314** [comment]
Thanks!

**Pull request merged**